iohk-monitoring/test/Cardano/BM/Test/Structured.lhs

```diff
@@ -34,15 +38,33 @@ tests = testGroup "Testing Structured Logging" [
 
 \subsubsection{Simple logging of text.}\label{code:logText}
 \begin{code}
+
+data Pet = Pet { name :: Text, age :: Int}
+           deriving (Show)
+
+instance ToJSON Pet where
+    toJSON (Pet n a) = 
+        object [ "kind" .= String "Pet"
+               , "name" .= toJSON n
+               , "age" .= toJSON a ]
+
+instance Transformable Text IO Pet where
+    trTransformer = trStructured
+    -- trTransformer tr = Tracer $ \arg ->
+    --     traceWith tr =<<
+    --         LogObject <$> pure ""
+    --                   <*> (mkLOMeta Debug Public)
+    --                   <*> pure (LogStructured $ encode arg)
+
 logText :: Assertion
 logText = do
-    cfg <- defaultConfigTesting
-    baseTrace :: Trace IO Text <- Setup.setupTrace (Right cfg) "logText"
-
-    let logTrace = toLogObject $ baseTrace
+    cfg <- defaultConfigStdout
+    baseTrace :: Tracer IO (LogObject Text) <- Setup.setupTrace (Right cfg) "logText"
 
-    traceWith logTrace "This is a simple message."
-    traceWith logTrace ".. and another!"
+    traceWith (toLogObject baseTrace) ("This is a simple message." :: Text)
```

Comment:
I"m wondering if there is different way to structure this example that is more in keeping with the expect use, ie describe the example as a function taking a tracer so the lines are
 > traceWith tr ....
and pass in (toLogobject tr 
would this illustrate the abstraction better ? 
People are going to cut and paste the examples - need to make sure that will lead them (generally) into good practices

Reply:
yes, I'll prepare a different one with more transformers and without Text or Pet. Maybe something existing..